### PR TITLE
Add eMMC life-time estimate support

### PR DESCRIPTION
**Pull request by agners**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to the supervisor)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [ ] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [ ] The code has been formatted using Black (`black --fast supervisor tests`)
- [x] Tests have been added to verify that the new code works.

If API endpoints of add-on configuration are added/changed:

- [ ] Documentation added/updated for [developers.home-assistant.io][docs-repository]

[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[docs-repository]: https://github.com/home-assistant/developers.home-assistant

### CODE REVIEW

**Comment by agners** [comment]
@pvizeli I'll rebase this. But does the general idea looks ok? Or should I create a separate class for the logic?

**Comment by pvizeli** [comment]
I think the code is fine, we can put it to the new hardware/helper.py module for now. I'm only not sure about the API ATTR name, but all other is fine 👍

**Comment by agners** [comment]
The naming is based on what they call it in the eMMC spec ("device life-time", plus the generic `ssd_` prefix). Maybe we can also add "est" or "estimate"?

image

File: supervisor/api/host.py [hunk 1]
```diff
@@ -43,6 +44,7 @@ async def info(self, request):
             ATTR_DISK_FREE: self.sys_host.info.free_space,
             ATTR_DISK_TOTAL: self.sys_host.info.total_space,
             ATTR_DISK_USED: self.sys_host.info.used_space,
+            ATTR_SSD_LIFE_TIME: self.sys_host.info.ssd_life_time,
```
Comment (pvizeli) [inline supervisor/api/host.py:47]: let's call it `ATTR_DISK_LIFE_TIME` which is aligned within `ATTR_DISK_USED` as the first start. We can later expose more and different data over the /os/ endpoint

Comment (frenck) [reply supervisor/api/host.py:47]: Suggestion (future proof) `ATTR_STORAGE_LIFE_TIME`

Comment (pvizeli) [reply supervisor/api/host.py:47]: It's more all other on that endpoint is called with DISK and means `/data`

File: supervisor/api/host.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
     ATTR_NAME,
     ATTR_OPERATING_SYSTEM,
     ATTR_SERVICES,
+    ATTR_SSD_LIFE_TIME,
```
Comment (pvizeli) [inline supervisor/api/host.py:22]:
```suggestion
    ATTR_DISK_LIFE_TIME,
```

File: supervisor/api/host.py (hunk 1 above)
Comment (pvizeli) [inline supervisor/api/host.py:47]:
```suggestion
            ATTR_DISK_LIFE_TIME: self.sys_host.info.ssd_life_time,
```

File: supervisor/const.py [hunk 3]
```diff
@@ -236,6 +236,7 @@
 ATTR_SOURCE = "source"
 ATTR_SQUASH = "squash"
 ATTR_SSID = "ssid"
+ATTR_SSD_LIFE_TIME = "ssd_life_time"
```
Comment (pvizeli) [inline supervisor/const.py:239]:
```suggestion
ATTR_DISK_LIFE_TIME = "disk_life_time"
```

File: supervisor/api/host.py (hunk 1 above)
Comment (agners) [reply supervisor/api/host.py:47]:
```suggestion
            ATTR_DISK_LIFE_TIME: self.sys_host.info.disk_life_time,
```

**Review by pvizeli — APPROVED** [review]

**Pull request merged**